src/main/java/com/aws/greengrass/detector/config/Config.java

```diff
@@ -7,50 +7,55 @@
 
 import com.aws.greengrass.componentmanager.KernelConfigResolver;
 import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
 import com.aws.greengrass.util.Coerce;
 
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 
+@SuppressWarnings("PMD.DataClass")
 public class Config {
     static final String INCLUDE_IPV4_LOOPBACK_ADDRESSES_CONFIG_KEY = "includeIPv4LoopbackAddrs";
     static final String INCLUDE_IPV4_LINK_LOCAL_ADDRESSES_CONFIG_KEY = "includeIPv4LinkLocalAddrs";
     static final int DEFAULT_MQTT_PORT = 8883;
     static final boolean DEFAULT_INCLUDE_IPV4_LOOPBACK_ADDRESSES = false;
     static final boolean DEFAULT_INCLUDE_IPV4_LINK_LOCAL_ADDRESSES = false;
 
-    private AtomicInteger mqttPort = new AtomicInteger(DEFAULT_MQTT_PORT);
-    private AtomicBoolean includeIPv4LoopbackAddrs = new AtomicBoolean(DEFAULT_INCLUDE_IPV4_LOOPBACK_ADDRESSES);
-    private AtomicBoolean includeIPv4LinkLocalAddrs = new AtomicBoolean(DEFAULT_INCLUDE_IPV4_LINK_LOCAL_ADDRESSES);
+    private final AtomicInteger mqttPort = new AtomicInteger(DEFAULT_MQTT_PORT);
+    private final AtomicBoolean includeIPv4LoopbackAddrs = new AtomicBoolean(DEFAULT_INCLUDE_IPV4_LOOPBACK_ADDRESSES);
+    private final Logger logger;
+    private final AtomicBoolean includeIPv4LinkLocalAddrs =
+            new AtomicBoolean(DEFAULT_INCLUDE_IPV4_LINK_LOCAL_ADDRESSES);
 
     /**
      * Config constructor.
      *
      * @param topics Root Configuration topic
      */
     public Config(Topics topics) {
+        this.logger = LogManager.getLogger(this.getClass()).createChild();
         Topics configurationTopics = topics.lookupTopics(KernelConfigResolver.CONFIGURATION_CONFIG_KEY);
-        configurationTopics.subscribe((whatHappened, node) -> {
-            // Hardcoding port for now till MQTT Broker is not publishing it.
-            this.mqttPort = new AtomicInteger(DEFAULT_MQTT_PORT);
+        if (configurationTopics != null) {
```

Comment:
This should throw an exception if `lookupTopics` fails. This null check results in a silent failure. I'd remove this.

Same comment as above - please back out most of these changes. Nothing is needed except reading port from config, and then we can add the log Sushant asked for